controllers/integrationpipeline/integrationpipeline_adapter.go

```diff
@@ -144,7 +158,12 @@ func (a *Adapter) EnsureEphemeralEnvironmentsCleanedUp() (controller.OperationRe
 func GetIntegrationPipelineRunStatus(adapterClient client.Client, ctx context.Context, pipelineRun *tektonv1beta1.PipelineRun) (intgteststat.IntegrationTestStatus, string, error) {
 	// Check if the pipelineRun finished from the condition of status
 	if !h.HasPipelineRunFinished(pipelineRun) {
-		return intgteststat.IntegrationTestStatusInProgress, fmt.Sprintf("Integration test is running as pipeline run '%s'", pipelineRun.Name), nil
+		// Mark the pipelineRun's status as "Deleted" if its not finished yet and is marked for deletion (with a non-nil deletionTimestamp)
+		if pipelineRun.GetDeletionTimestamp() != nil {
+			return intgteststat.IntegrationTestStatusDeleted, fmt.Sprintf("Integration test which is running as pipeline run '%s', has been deleted", pipelineRun.Name), nil
```

Comment:
If user deletes one old PLR, I think this will cause the `Deleted` status written into snapshot annotation, this will overwrites the correct status.

Reply:
Or once a PLR is marked as deleted by integration-service, it won't appear in PLR list?

Reply:
Oh, my bad, old PLR is invisible for users, they operate on RHTAP UI.

Reply:
I think this logic is there to handle the weird behavior that happens when the `pipelineRun` is still running (hasn't finished) and something attempts to delete it. It shouldn't take into account old PLRs that had finished.

Reply:
`Deleted` status is only added to the PLRs that are still running and have been marked for deletion with a non-nil timestamp (this can happen when a user deletes the PLR manually).

In the case of an old PLR, if a user manually deletes it, there can be 2 scenarios based on the current state of this old PLR:
* If this old PLR is in a "Running" state:
    * Now when a user deletes it, based on the current logic, the controller would change its state within the status annotation from "InProgress" to "Deleted".
    * I think that's what we would ideally want to happen.

* If this old PLR is Finished:
    * Now when a user deletes it, based on the current logic, the controller would not do anything.
    * Why? Because at this point the status annotation would contain the correct status for this PLR like "TestPassed" (or failed, etc).
    * This action would just simply delete the PLR from the cluster since it would not contain any finalizer as it was removed right when the PLR was finished in the first place.

PLease correct me if I'm wrong, thanks!

Reply:
I see now. Thank you all for the explanations!